### PR TITLE
Implements publication status, adds issues_found flag

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds the `issues_found` flag, as well as an `update_status` method that will be called via lifecycle hooks whenever a thesis, hold, or author record is saved. The update_status method also relies on two new inversion helper methods that allow for the five qualifying conditions to be checked via `[a,b,c,d,e].all?` logic.

There is a guard check in the method that returns early if the thesis has already been picked up by the publication workflow.

I've tried to write unit tests pretty thoroughly on this work, to make sure that changes are happening as we expect. Hopefully they are sufficient, so I welcome some pretty tight scrutiny on all of this.

#### Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/ETD-343

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

YES

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: test/models/thesis_test.rb [hunk 1]
```diff
@@ -572,4 +551,168 @@ class ThesisTest < ActiveSupport::TestCase
...
+      "hold_source" => HoldSource.first,
+      "status" => "active",
+    })
+    assert_equal true, hold.valid?
+    hold.save
+    thesis.reload
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Adding a new hold when a thesis is in "Pending publication" will change nothing' do
```
Comment (JPrevost) [inline test/models/thesis_test.rb:693]: Thoughts on whether these last two checks should also be done on a status of "published"?

**Review by JPrevost — APPROVED** [review]
This works well and has clear tests and code!

I made a couple of comments you can choose to address or not depending on your thoughts on them.

File: test/models/thesis_test.rb (hunk 1 above)
Comment (matt-bernhardt) [reply test/models/thesis_test.rb:693]: Yeah, I was wondering about that; I also wasn't sure whether tests relating to "published" should be handled when that method gets written. I can write them now while I'm thinking about it.

File: test/models/thesis_test.rb [hunk 2]
```diff
@@ -572,4 +551,168 @@ class ThesisTest < ActiveSupport::TestCase
...
+  test 'publication status defaults to not ready for publication' do
+    thesis = Thesis.new
+    assert thesis.publication_status == 'Not ready for publication'
+  end
+
+  test 'invalid without publication status' do
+    thesis = theses(:one)
+    assert thesis.valid?
+    thesis.publication_status = nil
+    thesis.save
```
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply test/models/thesis_test.rb:564]: Oh, okay - I can see which of those saves aren't needed.

**Comment by matt-bernhardt** [comment]
Okay @JPrevost - thanks for the suggestions on testing. I've added some extra tests, and took out some of those unneeded saves while making sure that the existing tests still passed.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**